File: src/main/java/no/ccat/dto/HarvestDataSource.java
package no.ccat.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.Data;

import java.util.HashMap;
import java.util.Map;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "id",
        "dataSourceType",
        "url",
        "publisherId",
        "description",
        "acceptHeaderValue",
        "dataType"
})
public class HarvestDataSource {
    @JsonProperty("id")
    private String id;
    @JsonProperty("dataSourceType")
    private String dataSourceType;
    @JsonProperty("url")
    private String url;
    @JsonProperty("publisherId")
    private String publisherId;
    @JsonProperty("description")
    private String description;
    @JsonProperty("acceptHeaderValue")
    private String acceptHeaderValue;
    @JsonProperty("dataType")
    private String dataType;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<>();
}

File: src/main/kotlin/no/ccat/dto/HarvestDataSource.kt
package no.ccat.dto

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("id", "dataSourceType", "url", "publisherId", "description", "acceptHeaderValue")
class HarvestDataSource {
    @JsonProperty("id")
    val id: String? = null
    @JsonProperty("dataSourceType")
    val dataSourceType: String? = null
    @JsonProperty("url")
    val url: String? = null
    @JsonProperty("publisherId")
    val publisherId: String? = null
    @JsonProperty("description")
    val description: String? = null
    @JsonProperty("acceptHeaderValue")
    val acceptHeaderValue: String? = null
    @JsonIgnore
    val additionalProperties: Map<String, Any> = HashMap()
}